docs/developer-guide/testing.md

```diff
@@ -23,6 +23,11 @@ To pass additional options to Goblint with `update_suite.rb`, use the `gobopt` e
 gobopt='--sets exp.privatization write+lock' ./scripts/update_suite.rb
 ```
 
+### Writing
+* If you introduce a new warning message, you need to manually add it to `./scripts/update_suite.rb` in the declaration at line 386, otherwise it won`t be recognized as a warning
+    * Example: ` when /Array out of bound/        then "warn"`
+* Add parameters to a regression test in the first line: `//PARAM: --set dbg.debug true`
+* Annotate lines inside the regression test with comments: `arr[9] = 10; //WARN` 
```

Comment:
Maybe "Annotating tests" would be a better title for this section? Also the option `--set dbg.debug true` should already be activated by default by the regression test script, so maybe choose another example.

Reply:
> Also the option `--set dbg.debug true` should already be activated by default by the regression test script, so maybe choose another example.

I thought that too here https://github.com/goblint/analyzer/pull/256#discussion_r660474588, but apparently it depends on some intricate conditions I'm not sure are necessary.

Reply:
won`t -> won't
Also, 'at line 386' is fragile, better use a permalink which can also show the code inline.